aiida_project/data/shell_fields.yaml

```
@@ -0,0 +1,39 @@
+bash:
+  config_file: .bashrc
+  init_lines: &bash_init_lines |
+    export $(grep -v '^#' {env_file} | xargs)
+    cda () {
+      source "$aiida_venv_dir/$1/bin/activate"
+      cd "$aiida_project_dir/$1"
+    }
+  activate: |
+    export AIIDA_PATH={env_file_path}
+    eval "$(_VERDI_COMPLETE=bash_source verdi)"
+  deactivate: &bash_deactivate |
+    unset AIIDA_PATH
+zsh:
+  config_file: .zshrc
+  init_lines: *bash_init_lines
+  activate: |
+    export AIIDA_PATH={env_file_path}
+    eval "$(_VERDI_COMPLETE=zsh_source verdi)"
+  deactivate: *bash_deactivate
+fish:
+  config_file: .config/fish/conf.d/aiida_project.fish
+  init_lines: |
+    set -gx VIRTUAL_ENV_DISABLE_PROMPT 1
+    for line in (cat {env_file_path})
+        set key (echo $line | cut -d= -f1)
+        set value (echo $line | cut -d= -f2 | sed "s/'//g")
+        set -gx $key $value
+    end
+    function cda
+        source "$aiida_venv_dir/$argv[1]/bin/activate.fish"
+        cd "$aiida_project_dir/$argv[1]"
+    end
+    funcsave -q cda
+  activate: |
+    set -Ux AIIDA_PATH {env_file_path}
```

@GeigerJ2 I think we have to remove the `U` here, no? Else the variable definition seems to be added to the `~/.config/fish/fish_variables` file and will be set for all shell instances.

```suggestion
    set -x AIIDA_PATH {env_file_path}
```

---

Ah, seems we should be using the `-g` "global" variable setting, not the "universal" one:

https://fishshell.com/docs/current/cmds/set.html

---

Yeah, you're right. Should be `-g`, not `-U` indeed!